src/libsystemd/sd-bus/bus-message.c

```diff
@@ -597,6 +597,26 @@ static sd_bus_message *message_new(sd_bus *bus, uint8_t type) {
         return m;
 }
 
+_public_ int sd_bus_message_new(
+                sd_bus *bus,
+                sd_bus_message **m,
+                uint8_t type) {
+
+        sd_bus_message *t;
+        int r;
+
+        assert_return(bus, -ENOTCONN);
+        assert_return(bus->state != BUS_UNSET, -ENOTCONN);
+        assert_return(m, -EINVAL);
+
+        t = message_new(bus, type);
+        if (!t)
+                return -ENOMEM;
```

Comment:
I think it would be nicer to just replace message_new by sd_bus_message_new(), and patch the few places where message_new() is called right now. That way we only have a single function for all callers.